waiter/src/waiter/handler.clj

```diff
@@ -350,18 +350,21 @@
                                   instance-count (reduce + (vals instance-counts-map))
                                   effective-service-description (service-id->service-description-fn service-id :effective? true)
                                   resource-usage (compute-resource-usage scheduler service-id instance-count)
-                                  service-metrics (get service-id->metrics service-id)]
+                                  service-metrics (get service-id->metrics service-id)
+                                  {:keys [deployment-error-message service-status-label]} (service/retrieve-service-status-and-deployment-error service-id global-state)]
                               (cond->
                                 {:instance-counts instance-counts-map
                                  :last-request-time (get service-metrics "last-request-time")
                                  :request-metrics (select-keys service-metrics ["outstanding" "total"])
                                  :resource-usage resource-usage
                                  :service-id service-id
                                  :service-description core-service-description
-                                 :status (service/retrieve-service-status-label service-id global-state)
+                                 :status service-status-label
```

Comment:
changing from a message to a map is a breaking change. I can't think of any of our clients off the top of my head that use status, but should we try to avoid this? Maybe a new field that is `deployment-error`

Reply:
This field has not changed, it continues to be a string. The map being returned is destructured above. 
Our integration tests already rely on this field and would have complained if the value had changed.

Reply:
oops I misread and thought you were putting the return value for the function `(service/retrieve-service-status-and-deployment-error)` as the `:status`. This looks good